### ISSUE TITLE
Investigate behavior of Randomizer.nextPosDouble

### ISSUE THREAD

**Issue by jducoeur** [scalatest/scalatest#1469]
See [this TODO](https://github.com/scalatest/scalatest/blob/37cc3fd66e28b6b5e538586cb9cfcd343d4e54e8/scalatest/src/main/scala/org/scalatest/prop/Randomizer.scala#L161), which calls out an extremely suspicious-looking line that seems to be eliminating values between 0.0 and 1.0.  Like the comment, I have no idea why this line exists.

**Comment by bvenners** [comment]
Yes looking at in now I wonder if it was a bad translation from the one for Long or Int:

https://github.com/scalatest/scalatest/blob/37cc3fd66e28b6b5e538586cb9cfcd343d4e54e8/scalatest/src/main/scala/org/scalatest/prop/Randomizer.scala#L94

At that point it could be 0, which is not positive, so I just return 1L in the Long case. OK. But in the Double case we check if it is less than or equal to 1 and add 1 to it, which is kind of similar in spirit to what we do with Longs but wrong. I am thinking we should just replace a 0.0 with Double.MinPositiveValue.

**Comment by jducoeur** [comment]
Makes sense to me -- do you want me to do that?

**Comment by bvenners** [comment]
Yes please go ahead and make that change.

**Comment by jducoeur** [comment]
Okay -- it'll be in the Randomizer checkin.
